components/board/genericlinux/board.go

```diff
@@ -236,8 +250,11 @@ func (b *sysfsBoard) AnalogReaderByName(name string) (board.AnalogReader, bool)
 }
 
 func (b *sysfsBoard) DigitalInterruptByName(name string) (board.DigitalInterrupt, bool) {
-	b.logger.Warn("Digital interrupts are not currently supported on sysfs boards.")
-	return nil, false
+	interrupt, ok := b.interrupts[name]
```

Comment:
I think so, but am open to disagreement. If any pin could spontaneously turn into an interrupt pin, we'd need a way to shut down its use as a (non-interrupt) GPIO or PWM pin. but since something might be using it as GPIO while something else has turned it into an interrupt, we'd need a way to shut down the interrupt and turn on the GPIO capability, and it sounds like a real pain. We can do it if someone really wants, but it feels to me like a lot of work to support a poor design, and if we instead said "you have to specify in your config file which pins are interrupts and which are not," the whole mess goes away.

Reply:
I tend to agree in principle, but I would like @Fahmina and/or @stevebriskin to weigh in

Reply:
Gautham and I talked with Fahmina. We _do_ need to keep parity with this "get or create" functionality because when people configure an encoder and specify pins in there, they don't expect to also need to configure those same pins in the board config. However, this can be done as a separate task: I'll make a new Jira ticket, put a TODO in this PR, and merge.